orangewidget/widget.py

```diff
@@ -638,21 +662,28 @@ def __info(self):
                 in_msg = MessagesWidget(
                     objectName="input-summary", visible=False,
                     defaultStyleSheet=css,
-                    sizePolicy=QSizePolicy(QSizePolicy.Preferred,
-                                           QSizePolicy.Preferred)
+                    sizePolicy=QSizePolicy(QSizePolicy.Fixed,
+                                           QSizePolicy.Fixed)
                 )
                 out_msg = MessagesWidget(
                     objectName="output-summary", visible=False,
                     defaultStyleSheet=css,
-                    sizePolicy=QSizePolicy(QSizePolicy.Preferred,
-                                           QSizePolicy.Preferred)
+                    sizePolicy=QSizePolicy(QSizePolicy.Fixed,
+                                           QSizePolicy.Fixed)
                 )
+
+                in_out_msg = sb.findChild(QWidget, "in-out-msg")
+
                 # Insert a separator if these are not the first elements
-                # TODO: This needs a better check.
-                if sb.findChildren(SimpleButton):
-                    sb.addWidget(QFrame(frameShape=QFrame.VLine))
-                sb.addWidget(in_msg)
-                sb.addWidget(out_msg)
+                buttons = sb.findChild(QWidget, "buttons")
+                if buttons.layout().count() != 0:
```

Comment:
`buttons` can be None here if `want_main_layout` is False (the creation of the `buttons` should probably be moved to `statusBar` though).